db_task2/dbOperation.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+function getElectiveByID($electiveID){
+    try{
+        $conn=connection();
+        $stmt = $conn->prepare("SELECT * FROM electives WHERE id=:idPlaceholder limit 1;");
+        $stmt->bindParam(':idPlaceholder', $electiveID);
+        $stmt->execute();
+        return $stmt->fetch(PDO::FETCH_ASSOC);
+    }
+    catch(PDOException $error) {
+        die($error->getMessage());
+    }
+}
+
+
+function updateElectiveByID($electiveID){
+    try{
+        $conn=connection();
+        $courseTitle = $_POST['title'];
+        $name = $_POST['name'];
+        $description = $_POST['description'];
+        $stmt=$conn->prepare("UPDATE electives SET title=:titlePlaceholder, description=:descriptionPlaceholder, lecturer=:lecturerPlaceholder WHERE id=:idPlaceholder;");
+        $stmt->bindParam(':idPlaceholder', $electiveID);
+        $stmt->bindParam(':titlePlaceholder', $courseTitle);
+        $stmt->bindParam(':descriptionPlaceholder', $description);
+        $stmt->bindParam(':lecturerPlaceholder', $name);
+        $stmt->execute();
+    }
+    catch(PDOException $error) {
+        die($error->getMessage());
+    }
+}
+
+function addElective (){
+    try{
+        $conn=connection();
+        $courseTitle = $_POST['title'];
+        $name = $_POST['name'];
+        $description = $_POST['description'];
+        $stmt = $conn->prepare("INSERT INTO electives (title, description, lecturer) VALUES (:courseTitle, :description, :name);");
+        $stmt->bindParam(':courseTitle', $courseTitle);
+        $stmt->bindParam(':description', $description);
+        $stmt->bindParam(':name', $name);
+        $stmt->execute();
+    }
+    catch(PDOException $error) {
+        die($error->getMessage());
```

Comment:
Here you are basically outputting to the user the raw message of the exception. Sometimes raw messages may contain sensitive information that we may not want to share with the end users.

Reply:
You may want to change it everywhere.